### PR TITLE
ci: Add Fedora to the CI environment

### PR DESCRIPTION
**Pull request by Conan-Kudo**
As Fedora ships updated GNOME components often earlier than anyone else,
it makes sense to test this to ensure that this continues to work.

### CODE REVIEW

**Comment by ximion** [comment]
Do we really need a sanitized build on Fedora as well? I've deliberately put it only on one OS because it's quite slow (and getting faster feedback from the CI is often useful).
Also, Fedora Noob question: How often does rawhide break due to e.g. dependency issues in unrelated packages? If it's rarely/never, then we can leave everything as-is, otherwise we may want to ignore failures for that configuration. I'd also be fine with just testing it out, if there's no definitive answer ^^
Otherwise the PR looks fine, thanks for creating it! :-)

File: tests/ci/Dockerfile-fedora-latest [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#
+# Docker file for AppStream CI tests on Fedora Rawhide
```
Comment (ximion) [inline tests/ci/Dockerfile-fedora-latest:2]: Fedora Latest ;-)

**Comment by Conan-Kudo** [comment]
> Do we really need a sanitized build on Fedora as well? I've deliberately put it only on one OS because it's quite slow (and getting faster feedback from the CI is often useful).

Fedora Rawhide has the literally newest GCC, which I figured would be quite valuable for this.

> Also, Fedora Noob question: How often does rawhide break due to e.g. dependency issues in unrelated packages? If it's rarely/never, then we can leave everything as-is, otherwise we may want to ignore failures for that configuration. I'd also be fine with just testing it out, if there's no definitive answer ^^

It's relatively rare these days. The main breakage is around Python rebases usually, but we've streamlined that to avoid that problem as much as possible.

> Otherwise the PR looks fine, thanks for creating it! :-)

👍🏾

File: tests/ci/Dockerfile-fedora-latest (hunk 1 above)
Comment (Conan-Kudo) [reply tests/ci/Dockerfile-fedora-latest]: Fixed!

**Comment by ximion** [comment]
Latest GCC probably includes the dev release sometimes... Which might not be the worst thing to test against, so let's give this a try!

**Pull request merged**